### PR TITLE
Fix chrony-setup dependency in NTP service

### LINKED ISSUE #6181: ntp did not start on 1 of 4 gimlets
I launched `a4x2` testbed environment with this build of omicron:
```
$ git branch -vvv
* main 945e75fc7 [origin/main] [sled-agent] Rename switch zone functions to be more explicit (#6163)
```

ntp started on 3 of 4 gimlets (g0, g1, g3), but did not start on 1 of 4 gimlets (g2).

**Summary:** Lack of ntp synchronization on gimlet g2 prevented the remainder of `a4x2` testbed environment from coming up, as it was waiting for 4 of 4 gimlets to become time synchronized.  After running `svcadm restart ntp` on g2 (and no other modifications to the testbed), g2 became time synchronized, and remainder of `a4x2` run proceeded as expected.  It could be that chrony never picked up the config in `/etc/inet/chrony.conf`, and there could be some sort of race between sled agent creating that file and chrony starting up.

**Details:**

g2:
```
root@oxz_ntp_62a6e97a:~# chronyc -n tracking
Reference ID    : 00000000 ()
Stratum         : 0
Ref time (UTC)  : Thu Jan 01 00:00:00 1970
System time     : 0.000000000 seconds fast of NTP time
Last offset     : +0.000000000 seconds
RMS offset      : 0.000000000 seconds
Frequency       : 0.000 ppm slow
Residual freq   : +0.000 ppm
Skew            : 0.000 ppm
Root delay      : 1.000000000 seconds
Root dispersion : 1.000000000 seconds
Update interval : 0.0 seconds
Leap status     : Not synchronised

root@oxz_ntp_62a6e97a:~# chronyc -n sources
MS Name/IP address         Stratum Poll Reach LastRx Last sample               
===============================================================================
```

Output of above commands on g3 had had expected output.

g2:
```
root@oxz_ntp_62a6e97a:~# cat /etc/inet/chrony.conf
#
# Configuration file for an internal NTP server - one which communicates with
# boundary NTP servers within the rack.
#

driftfile /var/lib/chrony/drift
ntsdumpdir /var/lib/chrony
dumpdir /var/lib/chrony
pidfile /var/run/chrony/chronyd.pid
logdir /var/log/chrony

log measurements statistics tracking

# makestep <threshold> <limit>
# We allow chrony to step the system clock up to three times on startup if we
# are more than a second out. Even though boundary NTP servers are configured
# with local reference mode, they will not appear as authoritative servers
# until they synchronize upstream. This configuration allows us to converge
# quickly once the boundary servers converge. After this the clock will advance
# monotonically forward.
makestep 1.0 3

# When a leap second occurs we slew the clock over approximately 37 seconds.
leapsecmode slew
maxslewrate 2708.333

server 074a84a2-f2c3-43d7-8812-af254b7ffb7b.host.control-plane.oxide.internal iburst minpoll 0 maxpoll 4
server 2d1079f1-0380-469d-b18c-8c461dde09af.host.control-plane.oxide.internal iburst minpoll 0 maxpoll 4
```

g2:
```
root@oxz_ntp_62a6e97a:~# host 074a84a2-f2c3-43d7-8812-af254b7ffb7b.host.control-plane.oxide.internal
074a84a2-f2c3-43d7-8812-af254b7ffb7b.host.control-plane.oxide.internal has IPv6 address fd00:1122:3344:101::c
074a84a2-f2c3-43d7-8812-af254b7ffb7b.host.control-plane.oxide.internal has IPv6 address fd00:1122:3344:101::c
074a84a2-f2c3-43d7-8812-af254b7ffb7b.host.control-plane.oxide.internal has IPv6 address fd00:1122:3344:101::c

root@oxz_ntp_62a6e97a:~# host 2d1079f1-0380-469d-b18c-8c461dde09af.host.control-plane.oxide.internal
2d1079f1-0380-469d-b18c-8c461dde09af.host.control-plane.oxide.internal has IPv6 address fd00:1122:3344:102::c
2d1079f1-0380-469d-b18c-8c461dde09af.host.control-plane.oxide.internal has IPv6 address fd00:1122:3344:102::c
2d1079f1-0380-469d-b18c-8c461dde09af.host.control-plane.oxide.internal has IPv6 address fd00:1122:3344:102::c
```

g2:
```
root@oxz_ntp_62a6e97a:~# ping fd00:1122:3344:101::c
fd00:1122:3344:101::c is alive

root@oxz_ntp_62a6e97a:~# ping fd00:1122:3344:102::c
fd00:1122:3344:102::c is alive
```

g2: (first command has no output, so try the second comand)
```
root@oxz_ntp_62a6e97a:~# cat $(svcs -L ntp)

root@g2:~# /opt/oxide/oxlog/oxlog logs -a oxz_ntp_62a6e97a-ccd1-441f-b3c6-da461c38d60f ntp
/pool/ext/b77d6efc-5610-4437-a240-7ee44ffda43b/crypt/zone/oxz_ntp_62a6e97a-ccd1-441f-b3c6-da461c38d60f/root/var/svc/log/oxide-ntp:default.log.0

root@g2:~# cat /pool/ext/b77d6efc-5610-4437-a240-7ee44ffda43b/crypt/zone/oxz_ntp_62a6e97a-ccd1-441f-b3c6-da461c38d60f/root/var/svc/log/oxide-ntp:default.log.0
[ Jul 29 23:41:52 Enabled. ]
[ Jul 29 23:41:52 Rereading configuration. ]
[ Jul 29 23:41:53 Rereading configuration. ]
[ Jul 29 23:42:03 Executing start method ("/usr/sbin/chronyd -d &"). ]
2024-07-29T23:42:03Z chronyd version 4.5 starting (+CMDMON +NTP +REFCLOCK -RTC +PRIVDROP -SCFILTER +SIGND +ASYNCDNS -NTS +SECHASH +IPV6 -DEBUG)
[ Jul 29 23:42:03 Method "start" exited with status 0. ]
```

g2: (the first command has no effect)
```
root@oxz_ntp_62a6e97a:~# chronyc online
200 OK

root@oxz_ntp_62a6e97a:~# chronyc -n tracking
Reference ID    : 00000000 ()
Stratum         : 0
Ref time (UTC)  : Thu Jan 01 00:00:00 1970
System time     : 0.000000000 seconds fast of NTP time
Last offset     : +0.000000000 seconds
RMS offset      : 0.000000000 seconds
Frequency       : 0.000 ppm slow
Residual freq   : +0.000 ppm
Skew            : 0.000 ppm
Root delay      : 1.000000000 seconds
Root dispersion : 1.000000000 seconds
Update interval : 0.0 seconds
Leap status     : Not synchronised
```

g2:
```
root@oxz_ntp_62a6e97a:~# ntpdate -d fd00:1122:3344:101::c
ntpdig: querying fd00:1122:3344:101::c (fd00:1122:3344:101::c)
org t1: ea52d57c.c8547000 rec t2: ea52d57d.050a38eb
xmt t3: ea52d57d.050e1172 dst t4: ea52d57c.c88ca000
org t1: 1722308348.782538 rec t2: 1722308349.019687
xmt t3: 1722308349.019746 dst t4: 1722308348.783396
rec-org t21: 0.237149  xmt-dst t34: 0.236350
{"time":"2024-07-30T02:59:09.19745+0000","offset":0.236749,"precision":0.000399,"host":"fd00:1122:3344:101::c","ip":"fd00:1122:3344:101::c","stratum":4,"leap":"no-leap","adjusted":false}

root@oxz_ntp_62a6e97a:~# ntpdate -d fd00:1122:3344:102::c
ntpdig: querying fd00:1122:3344:102::c (fd00:1122:3344:102::c)
org t1: ea52d5dd.97063000 rec t2: ea52d5dd.968b739d
xmt t3: ea52d5dd.9691f12c dst t4: ea52d5dd.9745a800
org t1: 1722308445.589938 rec t2: 1722308445.588065
xmt t3: 1722308445.588164 dst t4: 1722308445.590907
rec-org t21: -0.001873  xmt-dst t34: -0.002742
{"time":"2024-07-30T03:00:45.588164+0000","offset":-0.002308,"precision":0.000435,"host":"fd00:1122:3344:102::c","ip":"fd00:1122:3344:102::c","stratum":4,"leap":"no-leap","adjusted":false}
```

g2: `/var/log/chrony` unexpectedly does not exist, even though we specified it in `/etc/inet/chrony.conf` as shown above
```
root@oxz_ntp_62a6e97a:~# ls -l /var/log
total 3
-rw-------   1 root     sys            0 Jul 29 23:41 authlog
-rw-r--r--   1 root     sys            0 Jul 29 23:45 syslog
-rw-r--r--   1 root     sys            0 Jul 29 23:41 syslog.0
```

g2:
```
root@oxz_ntp_62a6e97a:~# pgrep -lf chronyd
 2977 /usr/sbin/chronyd -d
 2983 /usr/sbin/chronyd -d

root@oxz_ntp_62a6e97a:~# ptree 2977
2303   zsched
  2977   /usr/sbin/chronyd -d
    2983   /usr/sbin/chronyd -d

root@oxz_ntp_62a6e97a:~# pstack 2983
2983:	/usr/sbin/chronyd -d
 fffffc7feefda99a recvmsg  (6, 1482780, 8000)
 fffffc7fef3723de __xnet_recvmsg (6, 1482780, 0) + 1e
 000000000042af5b receive_messages.constprop.0 () + 13b
 000000000042b40e SCK_ReceiveMessage () + 1e
 0000000000447348 helper_main () + 38
 0000000000447b31 PRV_StartHelper () + f1
 0000000000434c1d SYS_Solaris_DropRoot () + 15d
 0000000000423b94 main () + 754
 000000000041c1c7 _start_crt () + 87
 000000000041c128 _start () + 18

root@oxz_ntp_62a6e97a:~# pstack 2977
2977:	/usr/sbin/chronyd -d
 fffffc7feefdb13a pollsys  (fffffc7fffdf8f30, 3, fffffc7fffdf9050, 0)
 fffffc7feef6e1f1 pselect (6, fffffc7fffdf9170, 0, 0, fffffc7fffdf9050, 0) + 1e1
 fffffc7feef6e640 select (6, fffffc7fffdf9170, 0, 0, fffffc7fffdf90d0) + 70
 0000000000429b78 SCH_MainLoop () + 2c8
 0000000000423a02 main () + 5c2
 000000000041c1c7 _start_crt () + 87
 000000000041c128 _start () + 18
```

g2: this fixed the issue of no time synchronization and no logs
```
root@oxz_ntp_62a6e97a:~# svcadm restart ntp

root@oxz_ntp_62a6e97a:~# chronyc -n tracking
Reference ID    : A7E319D5 (fd00:1122:3344:101::c)
Stratum         : 5
Ref time (UTC)  : Tue Jul 30 03:10:38 2024
System time     : 0.000499962 seconds fast of NTP time
Last offset     : +0.000679364 seconds
RMS offset      : 0.001442548 seconds
Frequency       : 74.455 ppm fast
Residual freq   : -306.560 ppm
Skew            : 897.772 ppm
Root delay      : 0.032223161 seconds
Root dispersion : 0.002641839 seconds
Update interval : 1.0 seconds
Leap status     : Normal

root@oxz_ntp_62a6e97a:~# ls -l /var/log/chrony/
total 99
-rw-r--r--   1 chrony   chrony     19233 Jul 30 03:12 measurements.log
-rw-r--r--   1 chrony   chrony     16041 Jul 30 03:12 statistics.log
-rw-r--r--   1 chrony   chrony      8933 Jul 30 03:12 tracking.log
```

### PR DESCRIPTION
**Pull request by citrus-it**
Fixes #6181

### CODE REVIEW

File: smf/ntp/manifest/manifest.xml [hunk 1]
```diff
@@ -41,10 +41,10 @@
 
   <dependency name='chrony-setup' grouping='require_all' restart_on='refresh'
     type='service'>
-  <service_fmri value='svc:/oxide/zone-network-setup:default' />
```
Comment (karencfv) [inline smf/ntp/manifest/manifest.xml:44]: Why is this dependency being removed? AFAIK all omicron services need to depend on the zone-network-setup service

**Review by karencfv — COMMENTED** [review]
Oof nice catch! Thanks for fixing this. Just have a question about the previous dependency, but otherwise looks fine to me.

**Review by karencfv — APPROVED** [review]
Never mind! Just saw that the dependency is still there in a line above the one you fixed 😅

**Comment by citrus-it** [comment]
> Never mind! Just saw that the dependency is still there in a line above the one you fixed 😅

Yeah, it was presumably a copy and paste of the original zone-setup dependency that got missed. @elaine-oxide did the work of discovering that chrony was sometimes starting before chrony-setup had completed in #6181

**Comment by karencfv** [comment]
Thanks @elaine-oxide for all that work! Sorry for the bug 😳

**Review by rcgoodfellow — APPROVED** [review]
LGTM. Thanks!

**Comment by elaine-oxide** [comment]
I had a successful launch of a4x2 with this build of omicron:
```
$ git branch -vvv
* andy/chrony-setup 91317aa9a [origin/andy/chrony-setup] Fix chrony-setup dependency in NTP service
  main              8d7e9f227 [origin/main] Update Rust crate assert_cmd to 2.0.15 (#6191)
```
Keeping in mind that the behavior is timing dependent, it is likely that the problem is fixed, since I previously encountered the problem twice in a row (before the fix), and this time I did not encounter it (after the fix).

**Comment by elaine-oxide** [comment]
> Thanks @elaine-oxide for all that work! Sorry for the bug 😳

No worries.  I learned a lot things I need to learn anyways by debugging.  So thank you @karencfv for a relatively nice bug, and thank you @citrus-it for a very quick fix. ;)

**Comment by citrus-it** [comment]
> I had a successful launch of a4x2 with this build of omicron:

Thanks for testing. I know it is not definitive given it's a race, but it's a good sign. I've merged this and we'll see if it crops up again.

**Comment by elaine-oxide** [comment]
I had another successful launch of a4x2 with this build of omicron:
```
$ git branch -vvv
  andy/chrony-setup 91317aa9a [origin/andy/chrony-setup] Fix chrony-setup dependency in NTP service
* main              9d811b75d [origin/main] Fix chrony-setup dependency in NTP service (#6192)
```
So we've had success twice in a row with this fix, first time in the branch, and second time in main. :)

**Pull request merged**